sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala

```diff
@@ -923,6 +924,15 @@ object RocksDBFileManagerMetrics {
   val EMPTY_METRICS = RocksDBFileManagerMetrics(0L, 0L, 0L, None)
 }
 
+/**
+ * Case class to keep track of column family info within checkpoint metadata.
+ * @param cfId - virtual column family id
+ * @param isInternal - whether the column family is internal or not
+ */
+case class ColumnFamilyInfo(
+    cfId: Short,
+    isInternal: Boolean)
```

Comment:
It'd be ideal to make clear which is the source of truth - we seem to use the naming convention (starting with `_`) to determine the internal column family. Is it always reliable or we should look at the flag? If we decide the source of truth for this, I'd say the other should not be used to determine whether the cf is internal or not.

Reply:
Done